We configure Postgres with --disable-rpath because for the most part we
want to leave it to ldconfig to determine where libraries are.  However,
for some reason the Perl package puts libperl.so in a nonstandard place
and doesn't add that place to the ldconfig search path.  I think this
is a Perl packaging bug, myself, but apparently it's not going to change.
So work around it by adding an rpath spec to plperl.so (only).
Per bug #162198.

--- postgresql-8.1.15/src/pl/plperl/GNUmakefile.old	2008-12-03 13:01:25.000000000 +0200
+++ postgresql-8.1.15/src/pl/plperl/GNUmakefile	2008-12-03 13:01:38.000000000 +0200
@@ -36,6 +36,9 @@
 
 SHLIB_LINK = $(perl_embed_ldflags) $(BE_DLLLIBS)
 
+# Force rpath to be used even though we disable it everywhere else
+SHLIB_LINK += $(rpath)
+
 REGRESS_OPTS = --dbname=$(PL_TESTDB) --load-language=plperl
 REGRESS = plperl plperl_trigger plperl_shared plperl_elog
 
